testsuite/MDAnalysisTests/auxiliary/test_xvg.py

```diff
@@ -81,28 +96,44 @@ def __init__(self):
         self.format = "XVG-F"
         self.description['format'] = self.format
 
-class TestXVGFileReader(TestXVGReader):
-    def __init__(self):
-        reference = XVGFileReference()
-        super(TestXVGReader, self).__init__(reference)
 
-    def test_get_auxreader_for(self):
+class TestXVGFileReader(TestXVGReader):
+    @staticmethod
+    @pytest.fixture()
+    def ref():
+        return XVGFileReference()
+
+    @staticmethod
+    @pytest.fixture()
+    def reader(ref):
+        return ref.reader(
+            ref.testdata,
+            initial_time=ref.initial_time,
+            dt=ref.dt,
+            auxname=ref.name,
+            time_selector=None,
+            data_selector=None
+        )
+
+    def test_get_auxreader_for(self, ref, reader):
         # Default reader of .xvg files is intead XVGReader, not XVGFileReader
         # so test specifying format 
-        reader = mda.auxiliary.core.get_auxreader_for(self.ref.testdata,
-                                                      format=self.ref.format)
-        assert_equal(reader, self.ref.reader)
+        reader = mda.auxiliary.core.get_auxreader_for(ref.testdata,
+                                                      format=ref.format)
+        assert_equal(reader, ref.reader)
 
-    def test_reopen(self):
-        self.reader._reopen()
+    def test_reopen(self, reader):
+        reader._reopen()
         # should start us back at before step 0, so next takes us to step 0
-        self.reader.next()
-        assert_equal(self.reader.step, 0)
+        reader.next()
+        assert_equal(reader.step, 0)
+
 
 def test_xvg_bz2():
     reader = mda.auxiliary.XVG.XVGReader(XVG_BZ2)
     assert_array_equal(reader.read_all_times(), np.array([0., 50., 100.]))
 
+
 def test_xvg_bz2():
```

Comment:
If you rename  the function so that the test is run as well the PR is good to go.